temporalcli/client.go

```diff
@@ -12,52 +10,175 @@ import (
+	// Load a client config profile
+	var clientProfile envconfig.ClientConfigProfile
+	if !cctx.RootCommand.DisableConfigFile || !cctx.RootCommand.DisableConfigEnv {
+		var err error
+		clientProfile, err = envconfig.LoadClientConfigProfile(envconfig.LoadClientConfigProfileOptions{
+			ConfigFilePath:    cctx.RootCommand.ConfigFile,
+			ConfigFileProfile: cctx.RootCommand.Profile,
+			DisableFile:       cctx.RootCommand.DisableConfigFile,
+			DisableEnv:        cctx.RootCommand.DisableConfigEnv,
+			EnvLookup:         cctx.Options.EnvLookup,
+		})
 		if err != nil {
-			return nil, fmt.Errorf("grpc-meta %s", err)
+			return nil, fmt.Errorf("failed loading client config: %w", err)
 		}
-		clientOptions.HeadersProvider = headers
 	}
 
-	// Remote codec
-	if c.CodecEndpoint != "" {
-		codecHeaders, err := NewStringMapHeaderProvider(c.CodecHeader)
+	// To support legacy TLS environment variables, if they are present, we will
+	// have them force-override anything loaded from existing file or env
+	if !cctx.RootCommand.DisableConfigEnv {
+		oldEnvTLSCert, _ := cctx.Options.EnvLookup.LookupEnv("TEMPORAL_TLS_CERT")
+		oldEnvTLSCertData, _ := cctx.Options.EnvLookup.LookupEnv("TEMPORAL_TLS_CERT_DATA")
+		oldEnvTLSKey, _ := cctx.Options.EnvLookup.LookupEnv("TEMPORAL_TLS_KEY")
+		oldEnvTLSKeyData, _ := cctx.Options.EnvLookup.LookupEnv("TEMPORAL_TLS_KEY_DATA")
+		oldEnvTLSCA, _ := cctx.Options.EnvLookup.LookupEnv("TEMPORAL_TLS_CA")
+		oldEnvTLSCAData, _ := cctx.Options.EnvLookup.LookupEnv("TEMPORAL_TLS_CA_DATA")
+		if oldEnvTLSCert != "" || oldEnvTLSCertData != "" ||
+			oldEnvTLSKey != "" || oldEnvTLSKeyData != "" ||
+			oldEnvTLSCA != "" || oldEnvTLSCAData != "" {
+			if clientProfile.TLS == nil {
+				clientProfile.TLS = &envconfig.ClientConfigTLS{}
+			}
+			if oldEnvTLSCert != "" {
+				clientProfile.TLS.ClientCertPath = oldEnvTLSCert
+			}
+			if oldEnvTLSCertData != "" {
+				clientProfile.TLS.ClientCertData = []byte(oldEnvTLSCertData)
+			}
+			if oldEnvTLSKey != "" {
+				clientProfile.TLS.ClientKeyPath = oldEnvTLSKey
+			}
+			if oldEnvTLSKeyData != "" {
+				clientProfile.TLS.ClientKeyData = []byte(oldEnvTLSKeyData)
+			}
+			if oldEnvTLSCA != "" {
+				clientProfile.TLS.ServerCACertPath = oldEnvTLSCA
+			}
+			if oldEnvTLSCAData != "" {
+				clientProfile.TLS.ServerCACertData = []byte(oldEnvTLSCAData)
+			}
+		}
+	}
+
+	// Override some values in client config profile that come from CLI args. Some
+	// flags, like address and namespace, have CLI defaults, but we don't want to
+	// override the profile version unless it was _explicitly_ set.
+	var addressExplicitlySet, namespaceExplicitlySet bool
+	if cctx.CurrentCommand != nil {
+		addressExplicitlySet = cctx.CurrentCommand.Flags().Changed("address")
+		namespaceExplicitlySet = cctx.CurrentCommand.Flags().Changed("namespace")
+	}
+	if addressExplicitlySet {
+		clientProfile.Address = c.Address
+	}
+	if namespaceExplicitlySet {
+		clientProfile.Namespace = c.Namespace
+	}
```

Comment:
```suggestion
	if namespaceExplicitlySet {
		clientProfile.Namespace = c.Namespace
	} else {
		c.Namespace = clientProfile.Namespace
	}
```

Some code is using `c.Namespace` flag in its output which is obviously not valid if it's overwritten by env config stuff. Need to either update `c.Namespace` with the valid value or have some other way to relay what is the now-current namespace value.

Reply:
Fixed